src/TestingUtils/Microsoft.AspNetCore.Testing/src/xunit/ConditionalTheoryDiscoverer.cs

```diff
@@ -14,6 +14,25 @@ public ConditionalTheoryDiscoverer(IMessageSink diagnosticMessageSink)
         {
         }
 
+        private sealed class OptionsWithPreEnumerationEnabled : ITestFrameworkDiscoveryOptions
```

Comment:
Why is this workaround required? Is there a bug in xunit or Arcade?

Reply:
I'm certainly not an XUnit expert but it seems to me like a bug in XUnit. There doesn't seem to be any way to skip a particular DataRow when `PreEnumerateTheories` is false. Unless this is "by design", in which-case our options would seem to be:

1. Do this workaround (or a similar one) to force enumeration in this scenario anyway.
2. Abandon this scenario all together and not have SkippableDataRows
3. Call it a known issue an move on.

I figure I'll file an issue against XUnit, a tracking issue on us and see where it gets us? Are we fine with leaving this in until we hear back from them?

Reply:
Yes, I'm fine leaving this as is for now.

Reply:
https://github.com/xunit/xunit/issues/1871

Reply:
https://github.com/aspnet/Extensions/issues/747